paasta_tools/tron_tools.py

```diff
@@ -999,10 +1001,11 @@ def format_tron_action_dict(action_config: TronActionConfig, use_k8s: bool = Fal
         if executor == "spark":
             # this service account will only be used by Spark drivers since executors don't
             # need Kubernetes access permissions
+            spark_iam_role = action_config.get_iam_role() or DEFAULT_SPARK_IAM_ROLE
             result["service_account_name"] = create_or_find_service_account_name(
-                iam_role=action_config.get_iam_role(),
+                iam_role=spark_iam_role,
                 namespace=EXECUTOR_TYPE_TO_NAMESPACE[executor],
-                k8s_role=_spark_k8s_role(),
+                # k8s_role=_spark_k8s_role(),
```

Comment:
- Point env variable to KUBECONFIG = /etc/kubernetes/spark.conf (being used in adhoc mode)
- Puppet kubeconfig to k8s agents to be available in pods
- Mount KUBECONFIG to pods (extra volumes)
- command: env include unset env variables for unsetting the env variables set by /etc/boto_cfg
e.g. --unset=AWS_ACCESS_KEY_ID --unset=AWS_SECRET_ACCESS_KEY